lib/cylc/config.py

```diff
@@ -17,6 +17,10 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """Parse and validate the suite definition file
 
+Suite context is exported to the environment, which may be used in
```

Comment:
Maybe explain here exactly what "suite context" means.

Reply:
Maybe I should move and re-write the comment I re-wrote and shifted under the method:
```
    def process_suite_env(self):
        # Set local values of variables to give suite context before parsing
        # config, i.e for template filters (Jinja2, python ...etc) and possibly
        # needed locally by event handlers. Potentially task-specific due to
        # different directory paths on different task hosts, however, they are
        # overridden by tasks prior to job submission:
        for var, val in [
                ('CYLC_SUITE_NAME', self.suite),
                ('CYLC_DEBUG', str(cylc.flags.debug).lower()),
                ('CYLC_VERBOSE', str(cylc.flags.verbose).lower()),
                ('CYLC_SUITE_DEF_PATH', self.fdir),
                ('CYLC_SUITE_RUN_DIR', self.run_dir),
                ('CYLC_SUITE_LOG_DIR', self.log_dir),
                ('CYLC_SUITE_WORK_DIR', self.work_dir),
                ('CYLC_SUITE_SHARE_DIR', self.share_dir)]:
            os.environ[var] = val
```